### PR TITLE
SSCS-11585 - Dropdown List for Language selection values in update to case data 

### PR DESCRIPTION
**Pull request by hajam09**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by amoslv1 — APPROVED** [review]

**Review by AndreiTrushkov — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java [hunk 1]
```diff
@@ -323,4 +325,12 @@ private static String resolvePostCode(SscsCaseData sscsCaseData) {
 
         return sscsCaseData.getAppeal().getAppellant().getAddress().getPostcode();
     }
+
+    private void updateHearingOptionLanguageFromSelectedList(SscsCaseData sscsCaseData) {
+        HearingOptions hearingOptions = sscsCaseData.getAppeal().getHearingOptions();
+        if (hearingOptions != null) {
+            DynamicList languagesList = hearingOptions.getLanguagesList();
+            hearingOptions.setLanguages(isNull(languagesList) ? "" : languagesList.getValue().getLabel());
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java:333]: Is passing an empty string desirable here? What needs to happen for the list to be null here? Is there not a default value we can fall back on instead?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToStartHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Mock
+    private VerbalLanguagesService verbalLanguagesService;
+
+    @InjectMocks
+    private CaseUpdatedAboutToStartHandler handler;
+
+    @Before
+    public void setUp() {
+        sscsCaseData = SscsCaseData.builder().appeal(Appeal.builder().build()).build();
```
Comment (DGordonV1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToStartHandlerTest.java:54]: Could separate the builders out, so Appeal appeal = Appeal.buil...etc then set the .appeal(appeal) for caseData. General rule to keep these separate and not nest them (even when they're simple/empty like this).

**Review by DGordonV1 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java (hunk 1 above)
Comment (hajam09) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java:333]: This is very similar to what's done during the adjournCase event. The original idea was to convert the text field to a drop-down field, but the work required way too many changes. So we decided to keep the original text field (but hide it from the user) and just set it to whatever value was chosen from the dropdown list. If a new case is created in the future, both the text field and dropdown will be empty or null. Besides, the language text field was optional, so setting it to an empty string should be fine.

**Review by DGordonV1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java (hunk 1 above)
Comment (DGordonV1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java:333]: Cool, good explanation, thanks.

**Review by ArunabhaChowdhury — APPROVED** [review]

**Review by benouaer — APPROVED** [review]

**Pull request merged**